File: leetcode/python/[136] Single Number.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
      d = { nums[0]: 1 }
      
      for i in range(1, len(nums)):
        if not nums[i] in d:
          d[nums[i]] = 1
        else:
          d[nums[i]] += 1
          
      for item in d:
        if d[item] == 1:
          return item

File: leetcode/js/[136] Single Number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
 
/*
-> store count in an object
-> then filter the value with counter 1
-> return that key
*/
var singleNumber = function(nums) {
    const counter = {}
    
    for(let i = 0; i < nums.length; i += 1) {
      if (counter[nums[i]]) {
        counter[nums[i]] += 1
      } else {
        counter[nums[i]] = 1
      }
    }
  
    return Object.keys(counter).filter(key => counter[key] === 1)
};
